sdk/xmake.lua

```diff
@@ -167,6 +169,17 @@ rule("cherimcu.privileged-compartment")
 		target:add("defines", "CHERIOT_AVOID_CAPRELOCS")
 	end)
 
+rule("cherimcu.privileged-library")
+	add_deps("cherimcu.component")
+	on_load(function (target)
+		-- Mark this target as a CHERIoT compartment.
+		target:set("cherimcu.type", "compartment")
+		target:set("extension", ".compartment")
```

Comment:
Shouldn't these be library?

Reply:
IIRC, it was easier in the loader to treat these like compartments that exported libcalls than it was to treat them like libraries that needed `.rodata` capabilities installed.  I might be / have been wrong.

Reply:
Changing `cherimcu.type` is easy enough, but changing the extension to `".library"`, which I wouldn't expect to make much of a difference, instead prompts the last link to say
```
ld.lld: error: Import table for compartment 'allocator' refers to address 0x800025DC, which is not a valid export
```
(and similarly for other compartments).  `llvm-objdump` of the intermediate object says
```
error: address range table at offset 0x0 has a premature terminator entry at offset 0x10
```

At a quick glance through LLVM, I didn't see why that would be happening.  Suggestions?

Reply:
I don't really like the reimplementation of the logic in the library rule, this rule should just adopt that one.  If you just delegate to the library rule, does it do the right thing?

Reply:
This just confuses me.  There is nothing anywhere in the toolchain that should depend on the file extension (except the linker script for the final firmware - did you update it there as well?).

Reply:
Yes, see https://github.com/microsoft/cheriot-rtos/pull/98/commits/058c8d79501380bf3e751c0fc2e00ea43976beb9#diff-b2382c519e7cfb672fc11e41afb5a0ac747603b3502b42163a3bf902e974efc3, but that might be wrong?

Reply:
All of the bits that say `cheriot.token_library.compartment` need to say `cheriot.token_library.library`.

Reply:
The error message is a bit surprising, but understandable, I think.  If a section is referred to but isn't mentioned in the linker script then it gets shoved at the end.  With the wrong name, these things will be in the wrong place and so the firmware will 'link' except that the bit that calculates the report does some additional sanity checking and tells you it's nonsense.